src/main/java/io/vertx/redis/impl/RedisConnection.java

```diff
@@ -248,10 +248,14 @@ void send(final Command<?> command) {
         break;
       case CONNECTING:
       case ERROR:
-        pending.add(command);
+        runOnContext(v -> {
+          pending.add(command);
```

Comment:
yes that's a possibility, however I'm wondering if we could not end up with commands executed in the wrong order...